src/GraphQL.Tests/Validation/InputFieldsAndArgumentsOfCorrectLengthTests.cs

```diff
@@ -0,0 +1,287 @@
+using System.Collections.Generic;
+using GraphQL.Validation.Rules;
+using Xunit;
+
+namespace GraphQL.Tests.Validation
+{
+    public class InputFieldsAndArgumentsOfCorrectLengthTests : ValidationTestBase<InputFieldsAndArgumentsOfCorrectLength, ValidationSchema>
+    {
+        // INPUT FIELD
+
+        [Fact]
+        public void good_literal_input_field_length()
+        {
+            ShouldPassRule(@"
+            {
+              complicatedArgs {
+                complexArgField(complexArg: { requiredField: true, stringField: ""aaaa"" })
+              }
+            }");
+        }
+
+        [Fact]
+        public void good_variable_input_field_length()
+        {
+            ShouldPassRule(_ =>
+            {
+                _.Query = @"query q($value: String)
+                {
+                  complicatedArgs {
+                    complexArgField(complexArg: { requiredField: true, stringField: $value })
+                  }
+                }";
+                _.Inputs = new Dictionary<string, object> { ["value"] = "aaaa" }.ToInputs();
+            });
+        }
+
+        [Fact]
+        public void below_min_literal_input_field_length()
+        {
+            ShouldFailRule(_ =>
+            {
+                _.Query = @"
+                {
+                  complicatedArgs {
+                    complexArgField(complexArg: { requiredField: true, stringField: ""aa"" })
+                  }
+                }";
+                _.Error(
+                   message: "ObjectField \"stringField\" has invalid length (2). Length must be in range [3, 7].",
+                   line: 4,
+                   column: 72);
+            });
+        }
+
+        [Fact]
+        public void below_min_variable_input_field_length()
+        {
+            ShouldFailRule(_ =>
+            {
+                _.Query = @"query q($value: String)
+                {
+                  complicatedArgs {
+                    complexArgField(complexArg: { requiredField: true, stringField: $value })
```

Comment:
There should be a test for:

```gql
query q($complex: ...)
{
  complicatedArgs {
    complexArgField(complexArg: $complex)
  }
}
```

Where the stringField within the complex field is "aa".  So far I don't see how that will work with the current implementation.

Reply:
You are right 👍 , fixed. See the new `IVariableVisitor` design.